### PR TITLE
Parse markdown files

### PR DESCRIPTION
**Pull request by f-roland**
This PR adds more plugin level properties from the manifest, and allows parsing markdown files for the guide & description fields.

### CODE REVIEW

**Comment by f-roland** [comment]
@ranmeirman you want an option for it too ? There’s already a `screenshot` Whitelisted property in the manifest - I chose to pull the image url from that.

If we use an option, it means we need to handle the upload on s3

File: lib/plugin.rb [hunk 1]
```diff
@@ -59,6 +59,13 @@ def request_params
       params["plugin[category]"] = @manifest["type"]
       params["plugin[external_identifier]"] = @identifier
       params["plugin[whitelisted_account_ids][]"] = @manifest["whitelisted_account_ids"]
+      params["plugin[guide]"] = @plugin_guide || ""
```
Comment (ranmeirman) [inline lib/plugin.rb]: why we need this? what will happen if you send `nil`?

**Comment by ranmeirman** [comment]
@f-roland readme and documentation should be updated as well

File: lib/plugin.rb (hunk 1 above)
Comment (f-roland) [reply lib/plugin.rb]: I admit I didn't test it as both javascript & ruby will treat empty string as a falsy value - so I thought this was safer than allowing the value to be nil

**Comment by f-roland** [comment]
@ranmeirman readme updated. 
latest version is `0.40.0` so no need to bump it further than `0.41.0`

File: lib/plugin.rb (hunk 1 above)
Comment (ranmeirman) [reply lib/plugin.rb]: not sure I understand why you keep it. When posting nil value on request the values will be ignored. Why do we want empty strings in the db?

**Pull request merged**